filesystem/fs_interface.go

```diff
@@ -0,0 +1,13 @@
+package filesystem
+
+type FileSystem interface {
```

Comment:
have you looked at the official go s3 interface? or maybe a hdfs go client interface?